tests/unit/module_utils/test_turbo_module.py

```diff
@@ -92,3 +94,44 @@ def test_connect(monkeypatch):
     turbo_socket = turbo_common.AnsibleTurboSocket(socket_path="/nowhere")
     assert turbo_socket.bind()
     mocked_socket.connect_assert_called_once_with("/nowhere")
+
+
+def test_expand_argument_specs_aliases():
+    argspec = {"foo": {"type": int, "aliases": ["bar"]}}
+    assert expand_argument_specs_aliases(argspec) == {
+        "foo": {"type": int, "aliases": ["bar"]},
+        "bar": {"type": int, "aliases": ["bar"]},
+    }
+
+
+def test_prepare_args():
+    argspec = {"foo": {"type": int}}
+    params = {"foo": 1}
+    assert prepare_args(argspec, params) == {"ANSIBLE_MODULE_ARGS": {"foo": 1}}
+
+
+def test_prepare_args_ignore_none():
+    argspec = {"foo": {"type": int}}
+    params = {"foo": None}
+    assert prepare_args(argspec, params) == {"ANSIBLE_MODULE_ARGS": {}}
+
+
+def test_prepare_args_subkey_freeform():
+    argspec = {"foo": {"type": dict, "default": {}}}
+    params = {"foo": {"bar": 1}}
+    assert prepare_args(argspec, params) == {"ANSIBLE_MODULE_ARGS": {"foo": {"bar": 1}}}
+
+
+def test_prepare_args_subkey_with_default():
+    argspec = {"foo": {"bar": {"default": 1}}}
+    params = {"foo": {"bar": 1}}
+    assert prepare_args(argspec, params) == {"ANSIBLE_MODULE_ARGS": {"foo": {}}}
+
+
+def test_prepare_args_dedup_aliases():
+    argspec = {
+        "foo": {"aliases": ["bar"], "type": int},
+        "bar": {"aliases": ["bar"], "type": int},
+    }
+    params = {"foo": 1, "bar": 1}
+    assert prepare_args(argspec, params) == {"ANSIBLE_MODULE_ARGS": {"foo": 1}}
```

Comment:
ansible is raising exception when both parameter and its alias are set.

``
[WARNING]: Both option foo and its alias bar are set.
``
is this still the case with this modification ?

Reply:
This is the problem we want to avoid. Since we expand the argument_spec, we can set the same parameter two times and the background  daemon will rise the warning.

Reply:
ok got it